src/language/branches.md

```diff
@@ -0,0 +1,295 @@
+} else if (isSnowing()) {
+  you.wearJacket();
+} else {
+  car.putTopDown();
+}
+```
+
+The statement conditions must be expressions
+that evaluate to [boolean values][boolean], nothing else.
+To learn how to use `if` in an expression context, see [Conditional expressions][]
+
+## If case
+
+Dart `if` statements support `case` clauses followed by a [case pattern][pattern]: 
+
+```dart
+if (pair case [int x, int y]) return Point(x, y);
+```
+
+If the case pattern (the list pattern `[int x, int y]` in the example above)
+matches the case expression (`pair`), then the branch (`return Point(x, y)`)
+executes with any variables the pattern defines in scope (`x` and `y`).
+
+Otherwise, the else branch is executed, if there is one:
+
+```dart 
+if (pair case [int x, int y]) {
+  print('Was coordinate array $x,$y');
+} else {
+  throw FormatException('Invalid coordinates.');
+}
+```
+
+If-case is a simple way to match and destructure against a
+_single_ pattern. To test a value against _multiple_ patterns, use [switch](#switch-case).
+
+## Switch case
+
+Switch cases (both [statements](#switch-statements) and [expressions](#switch-expressions))
+in Dart compare an object against one or more [case patterns][pattern]. 
+
+Case patterns allow control flow to
+either match and destructure the object being switched, or continue execution
+if the object doesn't match. They can be any kind of [refutable pattern][]:
+
+- **Constant patterns**: the object's value has to be equivalent to match.
+  ```dart
+  switch (number) {
+    case 1:           // Matches if number == 1
+      print("one");     
+  }
+  ```
+- **List**, **map**, or **record patterns**: the elements are destructured if matched.
+  ```dart
+  switch (obj) {
+    case [a, b]:                  // Matches if obj is a list with two elements
+      print("$a and $b match");    
+  }
+  ```  
+- **Variable patterns**: the matched value binds to a new variable.
+  ```dart
+  switch ((1, 2)) {
+    case (var a, var b): ...     // Matches if 1 and 2 can be assigned to a and b
+  }
+  ```  
+- **Logical** or **relational patterns**: the object matches if the comparison to a constant returns `true`.
+  ```dart
+  switch (code) {
+    case >= first && <= last: ...    
+    // Matches if code's value is between the constant values of first and last
+  }
+  ```  
+- **Null check** or **assert patterns**: the match is first on whether the object is not null, then on the value.
+  ```dart
+  switch (maybeString) {
+    case var s?:                 
+    // Matches if maybe is not null, and then against the inner pattern "var s"
+  }
+  ```  
+- **Object patterns**: the match exposes getters on the object type.
+  ```dart
+  switch (Shape shape) {
+    case Rect(width: var w, height: var h): ...
+    // Matches if shape is of type Rect, and then against the properties of Rect
+  }
+  ```  
+
+Values destructured by a case pattern become local variables, only available in the
+scope of that clause. 
```

Comment:
People might not have an idea about how that scope is determined. Perhaps:
```suggestion
Values destructured by a pattern in a case become local variables, only available in the
body of that case (or group of cases, when one body is shared by several cases). 
```

Reply:
I'm going to omit the parenthetical because I think "the body of a case", regardless of how that body ended up associated with that case, reasonably covers shared case bodies